core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -363,14 +363,14 @@ private void fillDimensionsAndMeasuresForTables(TableSchema tableSchema) {
           if (!columnSchema.isInvisible() && columnSchema.isSortColumn()) {
             this.numberOfSortColumns++;
           }
-          if (!columnSchema.getEncodingList().contains(Encoding.DICTIONARY)) {
+          if (columnSchema.getDataType() != DataTypes.DATE) {
```

Comment:
I think all the places we can remove below code also.
encodings.add(Encoding.DIRECT_DICTIONARY);
      encodings.add(Encoding.DICTIONARY);

Also can remove the carbonColumn.hasEncoding(Encoding.DIRECT_DICTIONARY) check from all the places.

so, can lookup the usages of Encoding.DIRECT_DICTIONARY, we can remove everywhere other than thrift to maintain the compatibility

Reply:
similar comment for Encoding.DICTIONARY also

Reply:
Following need to be removed later in another PR
encodings.add(Encoding.DIRECT_DICTIONARY);
encodings.add(Encoding.DICTIONARY);